includes/modules/searchandreplace/class-pb-search.php

```diff
@@ -37,7 +49,18 @@ function search_for_pattern( $search, $limit, $offset, $orderby ) {
 			if ( ! ini_get( 'safe_mode' ) ) {
 				set_time_limit( 0 );
 			}
-			return $this->find( '@' . preg_quote( $search, '@' ) . '@', $limit, $offset, $orderby );
+			if ( $this->regex ) {
+				// First test that the search and replace strings are valid regex
+				set_error_handler( array( &$this, 'regex_error' ) );
+				$valid = @preg_match( $search, '', $matches );
```

Comment:
The errors are already being handled via `set_error_handler` (will display the formatted error to the user instead of randomly placing it in the HTML). Removing the `@` doesn't seem to have any effect, but I'd rather keep it for two reasons:

* It makes it clear that potential errors won't generate any error text at that place.
* I got that code extract from https://github.com/johngodley/search-regex/blob/master/models/search.php and I suspect there might be cases where leaving this out would still lead to unwanted behaviour. I don't know for sure why it was put there and am worried I might break something I don't know anything about. (e.g. I had some php installations where errors would always appear on screen twice; maybe its for that case). 

But since it seems to be working without, please let me know if you still want it removed.